### ISSUE TITLE
Should this file also include AWS Root CA 1?

### ISSUE THREAD

**Issue by idyll** [voltone/aws_rds_castore#2]
Should cert store file also include: 

     https://www.amazontrust.com/repository/AmazonRootCA1.pem

I think it's used by Serverless Aurora v1 and v2?

The current file works fine for PostgreSQL on AWS but I haven't got it working yet for serverless aurora. I'm not sure if the issue is the missing root or if the problem is that serverless aurora uses wildcard certs or if it's something else.

**Comment by idyll** [comment]
Sigh. It might need it, but I can't tell because I think verifyhostname breaks because of the wildcard certs that Aurora uses.

**Comment by voltone** [comment]
The primary goal of this lib was to bundle the private root CAs used by AWS RDS. The `AmazonRootCA1` certificate is a public CA that is present in most recent OS/browser trust stores, so it should be supported the 'traditional' way: automatically by OTP 26, using `public_key:cacerts_get()` on OTP 25, and using trust stores from Hex packages in prior OTP versions.

I am not familiar with the requirements for SSL options to connect to Aurora Serverless, but we could consider adding a helper function for it if it proves difficult to achieve without the help from a library such as this.

Have you tried with the following ssl_options?

```elixir
[
  server_name_indication: 'db_host_name.as.charlist',
  customize_hostname_check: [
    match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
  ]
]
```

**Comment by idyll** [comment]
Yeah. My issue is that we use a small RDS DB for our sandbox environment that is not Serverless Aurora, but then use Serverless Aurora in higher environments.

Probably need to just use it everywhere but when we set this up we were trying to save a little money.

I think to work as things stand I think I need to have the combination of your suggestion above and a file that includes the public root as well as the private roots. 

(Maybe I'm wrong about this, my impression is that once you specify that file with the keys it will *only* look at that file and not validate the public certs – that's why I was thinking it should be added.)
